lib/protocol/packets/RowDataPacket.js

```diff
@@ -62,7 +62,11 @@ function typeCast(field, parser, timeZone, supportBigNumbers, bigNumberStrings,
     case Types.NEWDATE:
       var dateString = parser.parseLengthCodedString();
       if (dateStrings) {
-        return dateString;
+          if (field.type === Types.DATE) {
```

Comment:
I'm happy to include it. When I first started poking into this module I found `NEWDATE` in the constants: https://github.com/mysqljs/mysql/blob/master/lib/protocol/constants/types.js , but I couldn't find it in mysql docs. Today I find 2 references to it in current docs: http://dev.mysql.com/doc/connector-net/en/connector-net-ref-mysqlclient-mysqlcommandmembers.html 

> Newdate: Obsolete. Use Datetime or Date type.

http://dev.mysql.com/doc/connectors/en/connector-net-ref-mysqlclient-mysqlcommand.html

> Newdate: Obsolete. Use Datetime or Date type.

So unless there is a decision for this module to remove support for this obsolete Member/Type and it is removed from the `switch` block in this file, I think I should amend this PR to support it.